odl/space/fspace.py

```diff
@@ -534,20 +1085,157 @@ def __init__(self, fspace, fcall=None, fapply=None):
             The actual instruction for out-of-place evaluation.
             It must return an `FunctionSet.range` element or a
             `numpy.ndarray` of such (vectorized call).
-        fapply : `callable`, optional
-            The actual instruction for in-place evaluation.
-            Its first argument must be the `FunctionSet.range`
-            element or the array of such (vectorization) to which the
-            result is written.
-
-        *At least one of the arguments ``fcall`` and ``fapply`` must
-        be provided.*
+        vectorized : bool
+            Whether the function supports vectorized
+            evaluation.
         """
         if not isinstance(fspace, FunctionSpace):
             raise TypeError('function space {!r} not a `FunctionSpace` '
                             'instance.'.format(fspace))
 
-        super().__init__(fspace, fcall, fapply)
+        FunctionSetVector.__init__(self, fspace, fcall, vectorized=vectorized)
+        if self._call_has_out and not self._call_out_optional:
+            # Now we can use a default out-of-place implementation
+            dtype = float if fspace.field == RealNumbers() else complex
+
+            def oop_wrapper(func):
+                @wraps(func)
+                def wrapper(x, **kwargs):
+                    return func(self, dtype, x, **kwargs)
+                return wrapper
+
+            self._call_out_of_place = oop_wrapper(_default_out_of_place)
+
+    # Convenience functions using element() need to be adapted
+    # TODO: Check which ones can be removed
```

Comment:
Yes, that depends also largely on the decision we take on the "always vectorize" question. If we go that way, I guess all of those can be removed, except `__pow__` which is more general than the one in `LinearSpace`.